lib/ja_resource.ex

```diff
@@ -22,25 +22,20 @@ defmodule JaResource do
   """
   @callback repo() :: module
 
-  defmacro __using__(opts) do
+  defmacro __using__(_opts) do
     quote do
       @behaviour JaResource
-      unquote(JaResource.use_action_behaviours(opts))
       unquote(JaResource.default_repo)
-    end
-  end
 
-  @doc false
-  def use_action_behaviours(opts) do
-    available = [:index, :show, :create, :update, :delete]
-    allowed = (opts[:only] || available -- (opts[:except] || []))
-
-    quote bind_quoted: [allowed: allowed] do
-      if :index  in allowed, do: use JaResource.Index
-      if :show   in allowed, do: use JaResource.Show
-      if :create in allowed, do: use JaResource.Create
-      if :update in allowed, do: use JaResource.Update
-      if :delete in allowed, do: use JaResource.Delete
+      # TODO: Extract JaResource.Repo to own module
+      #use JaResourse.Repo
+      use JaResource.Index
+      use JaResource.Show
+      use JaResource.Create
+      use JaResource.Update
+      use JaResource.Delete
+
+      plug JaResource.Plug
```

Comment:
This will need to be moved out of this use and into the target controllers. I left it in here just so I could run/test this against existing code bases without changing them.

Actual usage will end up looking like:

```elixir
defmodule MyApp.ArticleController do
  use MyApp.Webb, :controller
  use JaResource
  plug JaResource.Plug, only: [:index, :show] # Maybe JaResource.Dispatch or something

  def custom_action(conn, params) do
    # Custom actions handled by phoenix normally
  end

  def delete(conn, params) do
    # Actions not in only (or not excluded) can be implemented normally.
  end

  def update(conn, params) do
    # Behaviors can be dispatched directly without the plug too.
    JaResource.Update.call(__MODULE__, conn)
  end
end
```